lib/brainstem/api_docs/formatters/markdown/endpoint_formatter.rb

```diff
@@ -83,36 +83,34 @@ def format_endpoint!
           # Formats each parameter.
           #
           def format_params!
-            return unless endpoint.root_param_keys.any?
+            return unless endpoint.params_configuration_tree.any?
 
             output << md_h5("Valid Parameters")
             output << md_ul do
-              endpoint.root_param_keys.inject("") do |buff, (root_param_name, child_keys)|
-                if child_keys.nil?
-                  buff += parameter_with_indent_level(
-                    root_param_name,
-                    endpoint.valid_params[root_param_name],
-                    0
-                  )
-                else
-                  text = md_inline_code(root_param_name) + "\n"
-
-                  child_keys.each do |param_name|
-                    text += parameter_with_indent_level(
-                      param_name,
-                      endpoint.valid_params[param_name],
-                      1
-                    )
-                  end
-
-                  buff << md_li(text)
-                end
-
+              endpoint.params_configuration_tree.inject("") do |buff, (param_name, param_config)|
```

Comment:
If we want to always return the initial object for the next iteration, `each_with_object` might be a better choice than `inject`. It will allow us to get rid of explicitly returning `buff` on line 92.